### PR TITLE
feat: check fuel-core dependency version

### PR DESCRIPTION
**Pull request by Salka1988**
Added a function that checks the `fuel-core` version in the published `fuels` crate against our current version. If they don't match, a message prompts the user to align the versions to avoid potential errors. 
At the moment, I'm only comparing the major and minor, and I left out the patch.

### Checklist
- [x] I have linked to any relevant issues.
- [ ] I have updated the documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary labels.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [ ] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: packages/fuels-test-helpers/src/utils.rs [hunk 1]
```diff
@@ -14,3 +18,180 @@ pub fn into_message_configs(messages: Vec<Message>) -> Vec<MessageConfig> {
...
+                };
+
+                let fuel_core_req = fuel_core_req_vec
+                    .split('.')
+                    .take(2)
+                    .collect::<Vec<&str>>()
+                    .join(".");
+
+                if fuel_core_req != used_fuel_version {
+                    eprintln!(
```
Comment (segfault-magnet) [inline packages/fuels-test-helpers/src/utils.rs:183]: We shouldn't spam `stderr`. Any message you wish to relay is best done through logging at the appropriate level.

**Review by segfault-magnet — COMMENTED** [review]
I'd dump the support for `fuel-core-lib` or, better yet, implement the suggestions I made in the comments regarding getting a version out of `fuel-core`. 

This will cover only the users that are testing locally. Those testing against the testnet will still continue to ask questions.

We can do one of the following:
1. implement some kind of heuristic to try and give helpful suggestions to the user in the case of a version mismatch. Something as simple as `if err.contains( 'no field in request' )` might go a long way towards reducing user confusion.
2. program some sort of compatibility check in the SDK with an endpoint on the node. We might look up the version returned from fuel core and cross-reference our own compatibility table. 

Although we have no such table, a simple script that walks backward through the git history of every release can extract which pairs are known to work together. That alongside relying a bit on `fuel-core` respecting semver (most of the time) might be good enough.

**Comment by iqdecay** [comment]
Waiting on some refactoring first before reviewing.

**Comment by Salka1988** [comment]
> Waiting on some refactoring first before reviewing.

My mistake. This was supposed to be a draft through which I would exchange some ideas with you.

**Comment by iqdecay** [comment]
@FuelLabs/sdk-rust what do we do with this

**Comment by Salka1988** [comment]
> @FuelLabs/sdk-rust what do we do with this

We need this
 https://github.com/FuelLabs/fuels-rs/pull/1067

**Pull request closed without merging**